pkg/syncer/status/status_controller.go

```diff
@@ -60,15 +65,37 @@ type Controller struct {
 }
 
 func NewStatusSyncer(syncerLogger logr.Logger, syncTargetClusterName logicalcluster.Name, syncTargetName, syncTargetKey string, advancedSchedulingEnabled bool,
-	upstreamClient kcpdynamic.ClusterInterface, downstreamClient dynamic.Interface, downstreamInformers dynamicinformer.DynamicSharedInformerFactory, syncerInformers resourcesync.SyncerInformerFactory, syncTargetUID types.UID) (*Controller, error) {
+	upstreamClient kcpdynamic.ClusterInterface, downstreamClient dynamic.Interface,
+	ddsifForUpstreamSyncer *ddsif.DiscoveringDynamicSharedInformerFactory,
+	ddsifForDownstream *ddsif.GenericDiscoveringDynamicSharedInformerFactory[cache.SharedIndexInformer, cache.GenericLister, informers.GenericInformer],
+	syncTargetUID types.UID) (*Controller, error) {
 	c := &Controller{
 		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
 
-		upstreamClient:            upstreamClient,
-		downstreamClient:          downstreamClient,
-		downstreamNamespaceLister: downstreamInformers.ForResource(schema.GroupVersionResource{Version: "v1", Resource: "namespaces"}).Lister(),
+		upstreamClient:   upstreamClient,
+		downstreamClient: downstreamClient,
+
+		getDownstreamLister: func(gvr schema.GroupVersionResource) (cache.GenericLister, error) {
```

Comment:
why these wrappers? Just use the ddsifs?

Reply:
Because I had initially planned to possibly change the tests in order to directly mock these functions.
Finally I didn't do it (didn't want to push too much in this PR) , but it's no excluded in the future if we want to move towards the typical unit test pattern used in other KCP controllers, and finally remove the use of the kube fake client in syncer unit tests.

OTOH as it is now, it allows encapsulating the check of the presence / syncing status of the listers, which makes the code in `process` easier to read IMO.